bundles/org.openhab.core.config.discovery.usbserial.windowsregistry/src/main/java/org/openhab/core/config/discovery/usbserial/windowsregistry/internal/User32Ex.java

```diff
@@ -0,0 +1,149 @@
+import com.sun.jna.platform.win32.WinBase;
+import com.sun.jna.win32.W32APIOptions;
+
+/**
+ * Extra {@code user32.dll} mappings not defined in {@link User32}.
+ *
+ * @author Ravi Nadahar - Initial contribution.
+ */
+public interface User32Ex extends User32 {
+
+    /** The instance. */
+    User32Ex INSTANCE = Native.load("user32", User32Ex.class, W32APIOptions.DEFAULT_OPTIONS);
+
+    int WAIT_OBJECT_0 = NTStatus.STATUS_WAIT_0 + 0;
+    int WAIT_ABANDONED_0 = NTStatus.STATUS_ABANDONED_WAIT_0 + 0;
+    int WAIT_ABANDONED = WAIT_ABANDONED_0;
+    int WAIT_TIMEOUT = 258;
+    int WAIT_FAILED = 0xFFFFFFFF;
+    int MAXIMUM_WAIT_OBJECTS = 64;
+
+    int PM_NOREMOVE = 0;
+    int PM_REMOVE = 1;
+    int PM_NOYIELD = 2;
+
+    /**
+     * A {@code WM_KEYUP}, {@code WM_KEYDOWN}, {@code WM_SYSKEYUP}, or {@code WM_SYSKEYDOWN} message is in the queue.
+     */
+    int QS_KEY = 0x0001;
+
+    /** A {@code WM_MOUSEMOVE} message is in the queue. */
+    int QS_MOUSEMOVE = 0x0002;
+
+    /** A mouse-button message ({@code WM_LBUTTONUP}, {@code WM_RBUTTONDOWN}, and so on). */
+    int QS_MOUSEBUTTON = 0x0004;
+
+    /**
+     * A posted message (other than those listed here) is in the queue.
+     * <p>
+     * This value is cleared when you call {@link #GetMessage(MSG, HWND, int, int)} or
+     * {@link #PeekMessage(MSG, HWND, int, int, int)}, whether or not you are filtering messages.
+     *
+     * @see {@link #PostMessage(HWND, int, WPARAM, LPARAM)}
+     */
+    int QS_POSTMESSAGE = 0x0008;
+
+    /** A {@code WM_TIMER} message is in the queue. */
+    int QS_TIMER = 0x0010;
+
+    /** A {@code WM_PAINT} message is in the queue. */
+    int QS_PAINT = 0x0020;
+
+    /**
+     * A message sent by another thread or application is in the queue.
+     *
+     * @see {@link #SendMessage(HWND, int, WPARAM, LPARAM)}.
+     */
+    int QS_SENDMESSAGE = 0x0040;
+
+    /** A {@code WM_HOTKEY} message is in the queue. */
+    int QS_HOTKEY = 0x0080;
+
+    /**
+     * A posted message (other than those listed here) is in the queue.
+     * <p>
+     * This value is cleared when you call {@link #GetMessage(MSG, HWND, int, int)} or
+     * {@link #PeekMessage(MSG, HWND, int, int, int)} without filtering messages.
+     *
+     * @see {@link #PostMessage(HWND, int, WPARAM, LPARAM)}
+     */
+    int QS_ALLPOSTMESSAGE = 0x0100;
+
+    /** Windows XP and newer: A raw input message is in the queue. For more information, see Raw Input. */
+    int QS_RAWINPUT = 0x0400;
+
+    /** Windows 8 and newer: A touch input message is in the queue. For more information, see Touch Input. */
+    int QS_TOUCH = 0x0800;
+
+    /** Windows 8 and newer: A pointer input message is in the queue. For more information, see Pointer Input. */
+    int QS_POINTER = 0x1000;
+
+    /**
+     * A {@code WM_MOUSEMOVE} message or mouse-button message ({@code WM_LBUTTONUP}, {@code WM_RBUTTONDOWN}, and so on).
+     */
+    final int QS_MOUSE = (QS_MOUSEMOVE | QS_MOUSEBUTTON);
```

Comment:
You mark this as `final` .. so perhaps all the above constants should be `final` too?

Reply:
It's just an oversight - I can remove it. It's an interface, so they are all implicitly `public static final`.